### PR TITLE
ci: Enable deepspeed unit tests

### PR DESCRIPTION
**Pull request by MikhailKardash**
## Description

Enable deepspeed unit tests now that we have a GPU runner for them.

## Test Plan

Check that tests pass in CI

## Checklist

- [X] Changes have been manually QA'd

## Ticket
MLG-514

### CODE REVIEW

File: harness/tests/experiment/fixtures/tf_keras_xor_model.py
Comment (garrett361) [inline harness/tests/experiment/fixtures/tf_keras_xor_model.py:1]: Just verifying my understanding: all the `tf_utils` bits are just being used for fake data generation, right?

File: harness/tests/experiment/integrations/test_deepspeed_trial.py [hunk 1]
```diff
@@ -35,7 +35,7 @@ def manual_init_distributed() -> Iterator[None]:
         del os.environ["DET_MANUAL_INIT_DISTRIBUTED"]
 
 
-@pytest.mark.skipif(not torch.cuda.is_available(), reason="no gpu available")
```
[the remark this answers is not in the document]
Comment (garrett361) [reply harness/tests/experiment/integrations/test_deepspeed_trial.py:38]: I am surprised this isn't handed by the `deepspeed` and `gpu` marks.

**Review by garrett361 — APPROVED** [review]
Looks good to me, to the extent that I understand our CI.

File: harness/tests/experiment/integrations/test_deepspeed_trial.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (MikhailKardash) [reply harness/tests/experiment/integrations/test_deepspeed_trial.py:38]: I see, this is for the `make -C harness test` workflow? I'll put that back in.

**Review by tayritenour — APPROVED** [review]
LGTM

**Pull request merged**